tests/unit/amo/components/TestCollection.js

```diff
@@ -0,0 +1,445 @@
+      errorHandlerId: errorHandler.id,
+      page,
+      slug,
+      user,
+    }));
+  });
+
+  it('does not dispatch any action when nothing has changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const wrapper = renderComponent({ store });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps();
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when location has not changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const location = { query: {} };
+
+    const wrapper = renderComponent({ location, store });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({ location });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when loading collection', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    store.dispatch(fetchCollection({
+      errorHandlerId: errorHandler.id,
+      slug,
+      user,
+    }));
+
+    fakeDispatch.reset();
+    renderComponent({ store });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when loading collection page', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    store.dispatch(fetchCollectionPage({
+      errorHandlerId: errorHandler.id,
+      page: 123,
+      slug,
+      user,
+    }));
+
+    fakeDispatch.reset();
+    renderComponent({ store });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when there is an error', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const wrapper = renderComponent({ store });
+
+    const id = 'error-handler-id';
+    const error = createApiError({
+      response: { status: 500 },
+      apiURL: 'https://some/api/endpoint',
+      jsonResponse: { message: 'Nope.' },
+    });
+    store.dispatch(setError({ id, error }));
+    const capturedError = store.getState().errors[id];
```

Comment:
This pattern pre-dates https://github.com/mozilla/addons-frontend/pull/3010 as well as `shallowUntilTarget()`.

I think you can do it just by dispatching an error which makes this a lot simpler.

```js
const errorHandler = new ErrorHandler({
  id: 'some-id',
  dispatch: store.dispatch,
});
errorHandler.handle(new Error('an unexpected error'));

// You can only stub out dispatch after handling the error.
const fakeDispatch = sinon.stub(store, 'dispatch');

wrapper.setProps({ errorHandler });
```

Reply:
Actually, if you are only doing this for updates then you can probably do it like this:

```js
const wrapper = render(...);

const errorHandler = wrapper.instance().props.errorHandler;
errorHandler.handle(new Error('an unexpected error'));

const fakeDispatch = sinon.stub(store, 'dispatch');

wrapper.setProps({ errorHandler });
```

Reply:
ah thanks. did not know this and I was pretty sure we merged code with the code I had here. The new code is definitely better though.